api/pkg/apis/v1alpha1/providers/stage/wait/wait.go

```diff
@@ -301,6 +301,20 @@ func (i *WaitStageProvider) Process(ctx context.Context, mgrContext contexts.Man
 			return outputs, false, nil
 		}
 		counter++
+		if i.Config.WaitCount == 0 && counter%10 == 0 {
```

Comment:
What if WaitCount is a very large integer? I think we can have this check regardless of WaitCount

Reply:
Added in the new version. Now we check activation deletion not only for waitcount = 0